### PR TITLE
fix(python): Add initialization support for python Timedeltas

### LINKED ISSUE #11540: Series initialisation doesn't work for numpy timedelta64[ms]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
In [57]: pl.Series(np.array([date(2020, 1, 1)], dtype='datetime64[ms]'))
Out[57]:
shape: (1,)
Series: '' [datetime[ms]]
[
        2020-01-01 00:00:00
]

In [58]: pl.Series(np.array([timedelta(1)], dtype='timedelta64[ms]'))
Out[58]:
shape: (1,)
Series: '' [o][object]
[
        86400000 milliseconds
]
```

### Log output

_No response_

### Issue description

Initialisation works for `datetime64[ms]` but not `timedelta64[ms]`

### Expected behavior

Initialisation should work for both or neither

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.7
Index type:          UInt32
Platform:            Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:              3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         2.2.1
connectorx:          <not installed>
deltalake:           0.9.0
fsspec:              2023.6.0
gevent:              <not installed>
matplotlib:          3.7.1
numpy:               1.25.1
openpyxl:            <not installed>
pandas:              2.1.1
pyarrow:             12.0.1
pydantic:            2.0.2
pyiceberg:           <not installed>
pyxlsb:              <not installed>
sqlalchemy:          2.0.20
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by c-peters**
fix #11540

### CODE REVIEW

File: py-polars/polars/series/series.py [hunk 1]
```diff
@@ -6821,20 +6821,22 @@ def struct(self) -> StructNameSpace:
...
 
-def _resolve_datetime_dtype(
-    dtype: PolarsDataType | None, ndtype: np.datetime64
+def _resolve_temporal_dtype(
+    dtype: PolarsDataType | None,
+    ndtype: np.dtypes.TimeDelta64DType | np.dtypes.DateTime64DType,
 ) -> PolarsDataType | None:
-    """Given polars/numpy datetime dtypes, resolve to an explicit unit."""
+    """Given polars/numpy temporal dtypes, resolve to an explicit unit."""
+    PolarsType = Duration if ndtype.type == np.timedelta64 else Datetime
```
Comment (MarcoGorelli) [inline py-polars/polars/series/series.py:6829]: could the resolution be preserved for durations too?

- np.timedelta64[ms] -> Duration('ms')
- np.timedelta64[us] -> Duration('us')
- (same for 'ns')
?

**Review by MarcoGorelli — COMMENTED** [review]
thanks for working on this!

File: py-polars/tests/unit/datatypes/test_temporal.py [hunk 2]
```diff
@@ -203,6 +203,17 @@ def test_from_pydatetime() -> None:
     assert s.dt[0] == dates[0]
 
 
+def test_from_numpy_timedelta() -> None:
+    s = pl.Series(
+        "name",
+        np.array([timedelta(days=1), timedelta(seconds=1)], dtype="timedelta64[ms]"),
+    )
+    assert s.dtype == pl.Duration
```
Comment (MarcoGorelli) [inline py-polars/tests/unit/datatypes/test_temporal.py:211]: could we check the time_unit of the result as well please?

could be a good use case for parametrisation, something like
```python
@pytest.mark.parametrize('time_unit', ['ms', 'us', 'ns'])
def test_from_numpy_timedelta(time_unit) -> None:
    s = pl.Series(
        "name",
        np.array([timedelta(days=1), timedelta(seconds=1)], dtype=f"timedelta64[{time_unit}]"),
    )
    assert s.dtype == pl.Duration(time_unit)
```

Comment (c-peters) [reply py-polars/tests/unit/datatypes/test_temporal.py:211]: Sure!

**Review by MarcoGorelli — APPROVED** [review]
nice one!

**Pull request merged**